tests/jobs/ahm_analysis/test_ahm_analysis.py

```diff
@@ -0,0 +1,418 @@
+                "SNAKE_OIL_PARAM:OP1_DIVERGENCE_SCALE": [0, 1, 2],
+                "SNAKE_OIL_PRES:BPR_138_PERSISTENCE": [0, 1, 2],
+                "LOG10_SNAKE_OIL_PARAM:OP1_PERSISTENCE": [0, 1, 2],
+            },
+            [
+                "SNAKE_OIL_PARAM:OP1_DIVERGENCE_SCALE",
+                "SNAKE_OIL_PARAM:OP1_OCTAVES",
+                "SNAKE_OIL_PARAM:OP1_PERSISTENCE",
+                "SNAKE_OIL_PRES:BPR_138_PERSISTENCE",
+            ],
+        ],
+    ],
+)
+def test_get_updated_parameters(prior_data, expected_result):
+    """test function creates a dataframe with all scalar parameters"""
+    prior_data = pd.DataFrame(prior_data)
+    scalar_parameters = ["SNAKE_OIL_PARAM", "SNAKE_OIL_PRES"]
+    p_keysf = ahmanalysis.get_updated_parameters(prior_data, scalar_parameters)
+    assert sorted(p_keysf) == sorted(expected_result)
+
+
+@pytest.mark.parametrize(
+    "prior_data_w",
+    [
+        {
+            "SNAKE_OIL_PARAM:OP1_PERSISTENCE": np.array([0, 1, 2]),
+            "SNAKE_OIL_PARAM:OP1_OCTAVES": np.array([0, 1, 2]),
+            "SNAKE_OIL_PARAM:OP1_DIVERGENCE_SCALE": np.array([0, 1, 2]),
+            "SNAKE_OIL_PARAM:OP1_OFFSET": np.array([[0, 0, 0], [0, 0, 0]]),
+            "SNAKE_OIL_PRES:BPR_138_PERSISTENCE": np.array([0, 1, 2]),
+        }
+    ],
+)
+def test_warning_get_updated_parameters(prior_data_w):
+    """test function creates a dataframe with all scalar parameters"""
+    expected_msg = (
+        "WARNING: Parameter SNAKE_OIL_PARAM:OP1_OFFSET defined several times."
+    )
+    scalar_parameters = ["SNAKE_OIL_PARAM", "SNAKE_OIL_PRES"]
+    with pytest.warns(UserWarning, match=expected_msg):
+        ahmanalysis.get_updated_parameters(prior_data_w, scalar_parameters)
+
+
+def create_facade(keys):
+    def side_effect(key):
+        return keys[key]
+
+    facade = MagicMock()
+    facade.get_data_key_for_obs_key.side_effect = side_effect
+    return facade
+
+
+@pytest.mark.parametrize(
+    "keys, expected_result",
+    [
+        pytest.param(
+            {"obs_key": "data_key"},
+            {"data_key": ["obs_key"]},
+            id="One data key and one obs key",
+        ),
+        pytest.param(
+            {"obs_key_1": "data_key", "obs_key_2": "data_key"},
+            {"data_key": ["obs_key_1", "obs_key_2"]},
+            id="Multiple obs per data key",
+        ),
+        pytest.param(
+            {"obs_key_1": "data_key_1", "obs_key_2": "data_key_2"},
+            {"data_key_1": ["obs_key_1"], "data_key_2": ["obs_key_2"]},
+            id="Single obs for multiple data key",
+        ),
+        pytest.param(
+            {
+                "obs_key_1": "data_key_1",
+                "obs_key_2": "data_key_2",
+                "obs_key_3": "data_key_2",
+            },
+            {"data_key_1": ["obs_key_1"], "data_key_2": ["obs_key_2", "obs_key_3"]},
+            id="Multiple obs for multiple data keys",
+        ),
+    ],
+)
+def test_group_obs_data_key(keys, expected_result):
```

Comment:
Should we include some parts with columns, e.g. `obs:key_1`, since that is part of the functionality as well?

Reply:
done